service/worker/scheduler/workflow.go

```diff
@@ -514,21 +530,21 @@ func (s *scheduler) checkConflict(token int64) error {
 	return errUpdateConflict
 }
 
+func (s *scheduler) updateTweakables() {
+	// Use MutableSideEffect so that we can change the defaults without breaking determinism.
+	get := func(ctx workflow.Context) interface{} { return currentTweakablePolicies }
+	eq := func(a, b interface{}) bool { return a.(tweakablePolicies) == b.(tweakablePolicies) }
```

Comment:
does the == operator do deep compare on each fields?

Reply:
it's a shallow compare: it doesn't go into slices or pointers but structs are compared field by field. actually I believe in this case even `a == b` would work here, since interfaces are compared by type and then value.

Reply:
Please at least do a manual test to verify that when you don't change the config, run the schedule multiple times with server restart in between them, and verify there is only one marker in history. Then change the config value, and run it again, and verify a new marker with new value is recorded.

Reply:
I tested those scenarios (using AlwaysAppendTimestamp) and everything works as expected: new marker only when setting is changed, and current value of setting takes effect, with no determinism errors